src/SimpleGraphs/simpledigraph.jl

```diff
@@ -0,0 +1,29 @@
+"""
+$(TYPEDEF)
+
+A type representing a directed graph with unweighted, non-multiple edges.
+
+# Fields
+
+$(TYPEDFIELDS)
+"""
+mutable struct SimpleDiGraph{T<:Integer} <: AbstractGraph{T,SimpleEdge{T}}
```

Comment:
Why is it mutable? I guess to allow in-place modification of `ne`?

Reply:
presumably but it's kinda sad

Reply:
Maybe `ne::Base.RefValue{Int}`?

Reply:
that was my thought at first, and I think I suggested it somewhere but got corrected, if only I could find the issue...

then again, maybe it's time to start fresh!

Reply:
I guess in newer versions of Julia you can mark `fadjlist` and `badjlist` as `const` so it may not matter conceptually either way. I think I found using `Base.RefValue` was a bit slower in practice compared to mutating a field of a `mutable`.